compiler/qsc_qasm3/src/compile.rs

```diff
@@ -0,0 +1,4364 @@
+                    paren_expr.syntax(),
+                ));
+                self.push_semantic_error(kind);
+                None
+            }
+        } else {
+            Some(1)
+        }
+    }
+
+    fn compile_paren_expr(
+        &mut self,
+        paren_expr: &oq3_syntax::ast::ParenExpr,
+    ) -> Option<QasmTypedExpr> {
+        let span = span_for_syntax_node(paren_expr.syntax());
+        let expr = paren_expr.expr()?;
+        let texpr = self.compile_expr(&expr)?;
+        let pexpr = ast_builder::wrap_expr_in_parens(texpr.expr, span);
+        Some(QasmTypedExpr {
+            ty: texpr.ty.clone(),
+            expr: pexpr,
+        })
+    }
+
+    #[allow(clippy::too_many_lines)]
+    fn compile_negated_literal_as_ty(
+        &mut self,
+        literal: &Literal,
+        ty: Option<&Type>,
+    ) -> Option<QasmTypedExpr> {
+        let span = span_for_syntax_node(literal.syntax());
+        match literal.kind() {
+            LiteralKind::IntNumber(value) => match ty {
+                Some(Type::Float(..)) => {
+                    let expr = self.compile_int_to_double_literal(&value, true, span)?;
+                    Some(QasmTypedExpr {
+                        ty: ty.expect("Expected type").clone(),
+                        expr,
+                    })
+                }
+                _ => Some(compile_intnumber_as_negated_int(&value, span)),
+            },
+            LiteralKind::FloatNumber(value) => match ty {
+                Some(Type::Int(..) | Type::UInt(..)) => {
+                    let value = value.value().expect("FloatNumber must have a value");
+                    #[allow(clippy::cast_possible_truncation)]
+                    let converted_value = value.trunc() as i64;
+                    #[allow(clippy::cast_precision_loss)]
+                    if (converted_value as f64 - value).abs() > f64::EPSILON {
+                        let span = span_for_syntax_node(literal.syntax());
+                        let kind = SemanticErrorKind::CastWouldCauseTruncation(
+                            "Float".to_string(),
+                            format!("{:?}", ty.expect("Expected type")),
+                            span,
+                        );
+                        self.push_semantic_error(kind);
+                        None
+                    } else {
+                        let expr = build_lit_int_expr(-converted_value, span);
+                        let ty = ty.expect("Expected type").clone();
+                        Some(QasmTypedExpr { ty, expr })
+                    }
+                }
+                _ => Some(compile_floatnumber_as_negated_double(&value, span)),
+            },
+            _ => {
+                self.push_unimplemented_error_message(
+                    "negated literal expressions",
+                    literal.syntax(),
+                );
+                None
+            }
+        }
+    }
+
+    #[allow(clippy::too_many_lines)]
```

Comment:
Same here, this function is only 55 lines long and the lint triggers when you have more than 100 lines.